pennylane/gradients/spsa_gradient.py

```diff
@@ -0,0 +1,573 @@
+    parameters with respect to its inputs. This estimator shifts all parameters
+    simultaneously and approximates the gradient based on these shifts and a
+    finite-difference method. This function is adapted to the new return system.
+
+    Args:
+        tape (pennylane.QNode or .QuantumTape): quantum tape or QNode to differentiate
+        argnum (int or list[int] or None): Trainable parameter indices to differentiate
+            with respect to. If not provided, the derivatives with respect to all
+            trainable parameters are returned.
+        h (float or tensor_like[float]): Step size for the finite-difference method
+            underlying the SPSA. Can be a tensor-like object
+            with as many entries as differentiated *gate* parameters
+        approx_order (int): The approximation order of the finite-difference method underlying
+            the SPSA gradient.
+        n (int): compute the :math:`n`-th derivative
+        strategy (str): The strategy of the underlying finite difference method. Must be one of
+            ``"forward"``, ``"center"``, or ``"backward"``.
+            For the ``"forward"`` strategy, the finite-difference shifts occur at the points
+            :math:`x_0, x_0+h, x_0+2h,\dots`, where :math:`h` is the stepsize ``h``.
+            The ``"backwards"`` strategy is similar, but in
+            reverse: :math:`x_0, x_0-h, x_0-2h, \dots`. Finally, the
+            ``"center"`` strategy results in shifts symmetric around the
+            unshifted point: :math:`\dots, x_0-2h, x_0-h, x_0, x_0+h, x_0+2h,\dots`.
+        f0 (tensor_like[float] or None): Output of the evaluated input tape in ``tape``. If
+            provided, and the gradient recipe contains an unshifted term, this value is used,
+            saving a quantum evaluation.
+        validate_params (bool): Whether to validate the tape parameters or not. If ``True``,
+            the ``Operation.grad_method`` attribute and the circuit structure will be analyzed
+            to determine if the trainable parameters support the finite-difference method,
+            inferring that they support SPSA as well.
+            If ``False``, the SPSA gradient method will be applied to all parameters without
+            checking.
+        shots (None, int, list[int], list[ShotTuple]): The device shots that will be used to
+            execute the tapes outputted by this transform. Note that this argument doesn't
+            influence the shots used for tape execution, but provides information
+            to the transform about the device shots and helps in determining if a shot
+            sequence was used to define the device shots for the new return types output system.
+        num_directions (int): Number of sampled simultaneous perturbation vectors. An estimate for
+            the gradient is computed for each vector using the underlying finite-difference
+            method, and afterwards all estimates are averaged.
+
```

Comment:
We are missing some documentation for the sampler

Reply:
Gave it a shot, let me know whether it is clear like this :)